parlai/core/torch_generator_agent.py

```diff
@@ -1611,7 +1714,22 @@ def select_paths(self, logprobs, prior_scores, current_length):
         # get the actual word id from residual of the same division
         tok_ids = best_idxs % voc_size
 
-        return (hyp_ids, tok_ids, best_scores)
+        tok_scores = torch.gather(logprobs, 1, tok_ids.unsqueeze(0)).view(-1)
```

Comment:
there is still a bug here, because we need to select along the right hypothesis IDs

i think you need something like `tok_scores = torch.index_select(logprobs, 0, hyp_ids).gather(1, tok_ids.unsqueeze(1))`, which first selects the logprobs along the hypothesis ids, and then gathers along the specific token ids within each hypothesis id

Reply:
Ah, right. Somehow this one got lost in the cracks. What you suggested makes sense to me (adding `.view(-1)` at the end though)